internal/command/jobs/printers.go

```diff
@@ -119,6 +125,79 @@ func printJobDetails(cfg *config.Job, out io.Writer, job *models.Job) error {
 	return nil
 }
 
+func waitForJob(ctx context.Context, cfg *config.JobSubmit, out io.Writer, jobName string) error {
+
+	fmt.Fprintf(out, "Waiting for job execution\n")
+
+	looped := false
+
+	retry := poll.
+		NewPoll().
+		WithDelay(TimeBetweenJobRefresh).
+		WithTimeout(cfg.Timeout)
+
+	lastCursor := ""
+
+	err := retry.Until(func() bool {
+		j, err := getJob(cfg.Job, jobName)
+		if err != nil {
+			fmt.Fprintf(out, "Error getting job: %s", err.Error())
+
+			// We want to keep retrying if the timeout has not been exceeded
+			return false
+		}
+
+		attempt := j.Status.Attempts[0]
+		switch attempt.Phase {
+		case "failed":
+			os.Exit(int(attempt.State.Failed.ExitCode))
+
+			return true
+
+		case "succeeded":
+			os.Exit(int(attempt.State.Succeeded.ExitCode))
+
+			return true
+		case "queued":
+			if looped {
+				fmt.Fprintf(out, "\033[1A\033[K")
+			}
+
+			fmt.Fprintf(out, "Queued on Job Runner Group %s\n", j.Spec.RunnerGroup)
+		case "running":
```

Comment:
When a run completes we are always adding an extra 10 sec delay before terminating the execution (because the logs will terminate and we will go thru the retry process before finding out that the job has completed).

I would get the job once again after `logs.ShowLogs()` has completed (independent from the result), and if the job is in a terminal state end the execution there. Something like:

```golang
case "running":
	if looped {
		fmt.Fprintf(out, "\033[1A\033[K")
	}

	lastCursor, err = logs.ShowLogs(ctx, cfg.API, out, jobName, cfg.Attach, lastCursor, 0)
	if err != nil {
		fmt.Fprintf(out, "Error getting logs: %s", err.Error())
	}

	if j, err = getJob(cfg.Job, jobName); err == nil {
		switch j.Status.Attempts[0].Phase {
		case "failed":
			os.Exit(int(attempt.State.Failed.ExitCode))
			return true
		case "succeeded":
			return true
		}
	}
```

Reply:
Agree, also i removed the "failed" and "succeeded" from the all cases, since that case will never happen